src/python-frontend/function_call_expr.cpp

```diff
@@ -108,6 +108,70 @@ exprt function_call_expr::build_constant_from_arg() const
     double arg_value = arg["value"].get<double>();
     arg["value"] = static_cast<int>(arg_value);
   }
+  else if (func_name == "chr")
+  {
+    int int_value = 0;
+
+    // Handle unary negative constant (e.g., -1)
+    if (arg.contains("_type") && arg["_type"] == "UnaryOp")
```

Comment:
Why has this to be handled here specifically?

Reply:
I'm happy to move this to another place. Would you have any suggestions?

Reply:
I don't understand the python frontend well enough, so I cannot suggest another place unfortunately.
It just felt a bit weird to handle this explicitly in this place and not in some more general place. But maybe that's perfectly fine for how the python frontend works.

Reply:
I'm refactoring the code here: https://github.com/esbmc/esbmc/pull/2417

If you have some suggestions, please let me know.

I'm still learning the Python frontend with @brcfarias :-)